docs/design/declaring_entities.md

```diff
@@ -66,36 +67,35 @@ In order to determine whether two redeclarations refer to the same entity, we
 apply the rules:
 
 -   Two named declarations _declare the same entity_ if they have the same scope
-    and the same name.
--   One declaration _redeclares_ another if they declare the same entity and the
-    second declaration appears after the first, including the case where the
-    first declaration is imported. In this case, the second declaration is said
-    to be a _redeclaration_ of the first.
--   Two declarations _differ_ if the sequence of tokens in the declaration
-    following the introducer keyword and the optional scope, up to the semicolon
-    or open brace, is different, except for `unused` modifiers on parameters.
--   The program is invalid if it contains two owning declarations of the same
-    entity that differ.
-    -   The non-owned `extern library` declarations will only use semantic
-        matching for redeclarations, not syntactic matching.
+    and the same name. This includes imported declarations.
+-   When two named declarations declare the same entity, the second is said to
+    be a _redeclaration_.
+-   Two owned declarations _differ_ if they don't syntactically match.
+    -   The non-owned `extern library` declarations only require semantic
+        matching.
+-   The program is invalid if it contains two declarations of the same entity
+    that differ.
```

Comment:
I think we need the definition of "differ" to include non-owned `extern library` declarations for the last rule here, which isn't specific to owned declarations. Perhaps something like:
```suggestion
-   Two owned declarations _differ_ if they don't syntactically match.
    -   Otherwise, if one is a non-owned `extern library` declaration, declarations
        differ if they don't match semantically.
-   The program is invalid if it contains two declarations of the same entity
    that differ.
```